clients/python-wrapper/lakefs/branch.py

```diff
@@ -236,12 +236,15 @@ def revert(self, reference_id: str, parent_number: Optional[int] = None) -> None
         elif parent_number <= 0:
             raise ValueError("parent_number must be a positive integer")
 
+        ref = reference if isinstance(reference, str) else reference.id
+
         with api_exception_handler():
-            return self._client.sdk_client.branches_api.revert_branch(
+            self._client.sdk_client.branches_api.revert_branch(
                 self._repo_id,
                 self._id,
-                lakefs_sdk.RevertCreation(ref=reference_id, parent_number=parent_number)
+                lakefs_sdk.RevertCreation(ref=ref, parent_number=parent_number)
             )
+            return self.head.get_commit()
```

Comment:
This is a race condition, it might not return the reverted branch.  revert_branch returns the reference I think, and we can get that commit from lakeFS rather than from head.

Reply:
You mean a lakefs SDK API call, right? I can do that (I'll likely just steal the code from `get_commit()`).